server/src/main/java/org/tctalent/server/service/db/impl/CandidateOccupationServiceImpl.java

```diff
@@ -171,10 +170,11 @@ public List<CandidateOccupation> updateCandidateOccupations(UpdateCandidateOccup
         /* Obtain logged in user for audit fields */
         User user = authService.getLoggedInUser()
                 .orElseThrow(() -> new InvalidSessionException("Not logged in"));
-        Candidate candidate = authService.getLoggedInCandidate();
-        if (candidate == null) {
-            throw new InvalidSessionException("Not logged in");
-        }
+
+        // Fetch updated candidate object from the DB to collect all data updates that may have been made since logging in.
+        Candidate candidate = candidateRepository.findById(user.getCandidate().getId())
+                .orElseThrow(() -> new InvalidSessionException("Not logged in"));
+
```

Comment:
For consistency, can you route this call through the candidate service instead of directly to the candidate repository?

There's a getCandidate(long id) method on CandidateService for this. The implementation sets a few transient fields that will otherwise be missed. Those fields are not especially relevant to what you're doing here, but going through the service is a good pattern for consistency and avoiding future oversights.

Reply:
No problem I replaced with that candidateService method